### DISCUSSION TITLE
Entertainment/education buildings upgrade conditions

### DISCUSSION THREAD

**Discussion by tmancini85** [Keriew/augustus#987] [Ideas]
Playing a little with Augustus 3.2.0, I tried to upgrade theaters, schools, libraries and academies, which, according to manual, it's depending on desirability.

My first assumption was that, as in vanilla Caesar 3 for markets, baths and the senate, you needed to reach the value of 30 for the northern building tile, but I was wrong.

For people like me also interested in the maths under the hood, these are the conditions I discovered, looking in the source code and testing a bit, to upgrade a building (please correct me if I'm wrong or I'm missing something):

- either tile counts (in vanilla it was only the northern for buildings and the most desirable tile for houses);
- most desirable tile values needed (according to /src/building/figure.c):
  - fountains evolve at 21, 41 and 61 (according to [here](https://caesar3.heavengames.com/cgi-bin/caeforumscgi/display.cgi?action=st&fn=2&tn=3600&st=6#post7), couldn't find the numbers in the source);
  - more than 30 for markets, baths and the senate (as in vanilla);
  - more than 40 for schools;
  - more than 45 for theater;
  - more than 50 for libraries;
  - more or equal than 60 for academies;
- employees criterion:
  - theaters and schools require employees to be upgraded, but they stay upgraded even if they lose access to labour force;
  - libraries, academies, markets, baths and the senate don't require employees to upgrade (as in vanilla).

I don't know if and how upgraded theaters, schools, libraries and academies without employees affect culture rating.

Anyway, I like this upgrade possibility very much, thanks a lot for your efforts! :partying_face: 

My hints:

- decrease the value for the most desirable tile:
  - if not more than 30 as in vanilla, maybe not worse than "more than 40"; high values are hard to reach and can lead to a "statues randomly build and destroy" strategy to gain the upgrade;
- as a drawback for the the upgrade, increase the employees for an upgraded building, like 40-50%:
- increased coverage should be related to employees increase, and in any case should not offer more than 50-60% of the initial coverage:
  - in particular, 1200 people, instead of 500 for a standard theater, seems a bit too advantageous :wink: .
 
What do you think?

**Comment by crudelios** [comment]
> I don't know if and how upgraded theaters, schools, libraries and academies without employees affect culture rating.

I believe it's the same effect as their non-upgraded variants, meaning they won't count.

> decrease the value for the most desirable tile

This has to do with the fact that generally Academies are already placed on desirable areas, i e next to your villa blocks. Reducing desirability requirement to 30 would make most of them upgrade with little effort. 

> as a drawback for the the upgrade, increase the employees for an upgraded building, like 40-50%

We discussed that and figured it would be a mess. You'd get workforce changes simply due to the desirability level, which isn't easy to control. Generally we want desirability = good and adding such a trade-off could imbalance that aspect of the game.

> in particular, 1200 people, instead of 500 for a standard theater, seems a bit too advantageous

You may be correct there, but the main idea was to reduce the spam of culture buildings. Theaters were particularly bad in that regard.

**Comment by tmancini85** [comment]
Thanks a lot for your reply. I'm still convinced that some tuning would be useful, but I appreciate a lot your effort to follow reasonable criteria.

I'm also happy to get the chance to congratulate for version 4.0: I haven't occasion to play it yet, but I read the manual, and the improvements are impressive. Well done!

(Off topic: have you had a chance to see https://github.com/Keriew/augustus/issues/842#issuecomment-1795993209? Thanks in advance).
